sdk/core/azure-core/src/http/url.cpp

```diff
@@ -103,86 +103,21 @@ std::string Url::Decode(const std::string& value)
   return decodedValue;
 }
 
-static const char* unreserved
-    = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~";
-static const char* subdelimiters = "!$&'()*+,;=";
-
-std::string Url::EncodeHost(const std::string& host)
-{
-  return EncodeImpl(host, [](int c) { return c > 127; });
-}
-
-std::string Url::EncodePath(const std::string& path)
-{
-  const static std::vector<bool> shouldEncodeTable = []() {
-    const std::string pathCharacters
-        = std::string(unreserved) + std::string(subdelimiters) + "%/:@";
-
-    std::vector<bool> ret(256, true);
-    for (char c : pathCharacters)
-    {
-      ret[c] = false;
-    }
-    // we also encode % and +
-    ret['%'] = true;
-    ret['+'] = true;
-    return ret;
-  }();
-
-  return EncodeImpl(path, [](int c) { return shouldEncodeTable[c]; });
-}
-
-std::string Url::EncodeQuery(const std::string& query)
-{
-  const static std::vector<bool> shouldEncodeTable = []() {
-    std::string queryCharacters = std::string(unreserved) + std::string(subdelimiters) + "%/:@?";
-
-    std::vector<bool> ret(256, true);
-    for (char c : queryCharacters)
-    {
-      ret[c] = false;
-    }
-    // we also encode % and +
-    ret['%'] = true;
-    ret['+'] = true;
-    // Surprisingly, '=' also needs to be encoded because Azure Storage server side is so strict.
-    // We are applying this function to query key and value respectively, so this won't affect
-    // that = used to separate key and query.
-    ret['='] = true;
-    return ret;
-  }();
-
-  return EncodeImpl(query, [](int c) { return shouldEncodeTable[c]; });
-}
-
-std::string Url::EncodeFragment(const std::string& fragment)
-{
-  const static std::vector<bool> shouldEncodeTable = []() {
-    std::string queryCharacters = std::string(unreserved) + std::string(subdelimiters) + "%/:@?";
-
-    std::vector<bool> ret(256, true);
-    for (char c : queryCharacters)
-    {
-      ret[c] = false;
-    }
-    // we also encode % and +
-    ret['%'] = true;
-    ret['+'] = true;
-    return ret;
-  }();
-
-  return EncodeImpl(fragment, [](int c) { return shouldEncodeTable[c]; });
-}
-
-std::string Url::EncodeImpl(const std::string& source, const std::function<bool(int)>& shouldEncode)
+std::string Url::Encode(const std::string& value, const char* extraSymbolsToEncode)
```

Comment:
why pass `const char*` instead of `const std::string&`?

Reply:
I saw you define your constant string in `Details::` namespace as 
```cpp
constexpr static const char* constant = "";
```

So I created this arg in a way you can create a constant for the chars you might want to use without creating std::string

Reply:
I do so because we cannot define a constexpr std::string

Reply:
So would you prefer `std::string&` ?  I can change if that works better for you.  I was trying to save you from building the std::string from a Details::constant

Reply:
I think consistency is the first thing we should follow. We hardly ever pass `const char*` as parameter in this C++ sdk.